src/os/bluestore/BlockDevice.h

```diff
@@ -148,6 +148,13 @@ class BlockDevice {
   virtual int invalidate_cache(uint64_t off, uint64_t len) = 0;
   virtual int open(const std::string& path) = 0;
   virtual void close() = 0;
+  bool is_valid_io(uint64_t off, uint64_t len) {
```

Comment:
done, BlockDevice::is_valid_io is called by derived class, so I put it under protected. please have a look.